### PR TITLE
feat!: support react 17 and upgrade react-redux to v7

### PR DESCRIPTION
**Pull request by laetitiadufau**
react-redux@7 is the first version to allow react@17 in its peer dependencies
BREAKING CHANGE: needs react-redux@7 to work

### CODE REVIEW

**Comment by laetitiadufau** [comment]
BREAKING CHANGES

**Review by lbdremy — APPROVED** [review]

File: src/ReduxRouter.js [hunk 1]
```diff
@@ -24,22 +24,28 @@ function getRoutesFromProps(props) {
...
-  }
-
-  static contextTypes = {
-    store: PropTypes.object
-  }
+    children: PropTypes.node,
+  };
 
   constructor(props, context) {
     super(props, context);
```
Comment (lbdremy) [inline src/ReduxRouter.js]: context is legacy

File: src/ReduxRouter.js [hunk 2]
```diff
@@ -24,22 +24,28 @@ function getRoutesFromProps(props) {
   return props.routes || props.children;
 }
 
+function withStore(ClassComponent) {
+  return function WrappedComponent(props) {
```
Comment (lbdremy) [inline src/ReduxRouter.js]: it is important to have a proper displayName to display correctly in react dev tools

e.g:

```
function withSubscription(WrappedComponent) {
  class WithSubscription extends React.Component {/* ... */}
  WithSubscription.displayName = `WithSubscription(${getDisplayName(WrappedComponent)})`;
  return WithSubscription;
}

function getDisplayName(WrappedComponent) {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
}
```

**Review by lbdremy — CHANGES_REQUESTED** [review]
See comments

**Review by lbdremy — APPROVED** [review]

**Pull request merged**